### DISCUSSION TITLE
how can i create a serializer for href

### DISCUSSION THREAD

**Discussion by Trassive** [readium/kotlin-toolkit#560] [Q&A] [answered]
i am using type safe arg for that my data class needs to be serializable how can i make Href serializable i have seperate screen for table of content so i want to pass the href to jump on specific part
```
@Serializable
    data class Reader(
        val id: Int,
        val href: Href
    ): LeafScreen
```
moreover is there any way to avoid false positive ( like a jump from one chapter to another) while saving the reading progress using currentLocator

**Comment by mickael-menu** [comment]
Which version of the Kotlin toolkit are you using? `Href` is different in 2.x and 3.x.

> moreover is there any way to avoid false positive ( like a jump from one chapter to another) while saving the reading progress using currentLocator

I don't understand, what do you mean by false positive in this context?

**Reply by Trassive** [reply]
i am using "3.0.0-beta.2" 
the second question was independent, apologies for not being clear let me explain again
i want to store the current locator to keep the track of last read location but how to avoid false positive meaning if by intentionally or unintentionally user scrolls many pages in that scenario i don't want to store that location

**Reply by mickael-menu** [reply, marked as answer]
I think there's a confusion with the `Href` in the v3, we'll explore an alternative. But basically you should not use `Href` here. Instead, use an `Url` from `Link.url()`. As for `Serializable`, `Url` is `Parcelable` so you might be able to write a serializer based on it. Or as an alternative, just use a `String` (`url.toString()` and `Url(string)` if you need to construct it again).

> i want to store the current locator to keep the track of last read location but how to avoid false positive meaning if by intentionally or unintentionally user scrolls many pages in that scenario i don't want to store that location

I wouldn't recommend doing that. We typically just store the current locator as is. Users may legitimately want to jump to a different location and have it become the new "last read location".
If you really want to pursue that path, there are no simple solutions. But you could check that the `locator.locations.position` is close to the previous one (not exactly +1, as you can have several positions in a single screen).

**Reply by Trassive** [reply]
Oh okay i get it thanks
